src/music.cpp

```diff
@@ -22,55 +22,50 @@
  * Read the name of a music CAT file entry.
  * @param filename Name of CAT file to read from
  * @param entrynum Index of entry whose name to read
- * @return Pointer to string, caller is responsible for freeing memory,
- *         nullptr if entrynum does not exist.
+ * @return Name of CAT file entry if it could be read.
  */
-char *GetMusicCatEntryName(const std::string &filename, size_t entrynum)
+std::optional<std::string> GetMusicCatEntryName(const std::string &filename, size_t entrynum)
 {
-	if (!FioCheckFileExists(filename, BASESET_DIR)) return nullptr;
+	if (!FioCheckFileExists(filename, BASESET_DIR)) return std::nullopt;
 
 	RandomAccessFile file(filename, BASESET_DIR);
 	uint32_t ofs = file.ReadDword();
 	size_t entry_count = ofs / 8;
-	if (entrynum < entry_count) {
-		file.SeekTo(entrynum * 8, SEEK_SET);
-		file.SeekTo(file.ReadDword(), SEEK_SET);
-		uint8_t namelen = file.ReadByte();
-		char *name = MallocT<char>(namelen + 1);
-		file.ReadBlock(name, namelen);
-		name[namelen] = '\0';
-		return name;
-	}
-	return nullptr;
+	if (entrynum >= entry_count) return std::nullopt;
+
+	file.SeekTo(entrynum * 8, SEEK_SET);
+	file.SeekTo(file.ReadDword(), SEEK_SET);
+	uint8_t namelen = file.ReadByte();
+
+	std::string name(namelen, '\0');
+	file.ReadBlock(name.data(), namelen);
+	return name;
```

Comment:
Probably not a big issue, but if `name` contains a `'\0'` somewhere internally you'd have a '\0' in the middle of a string. Previously that would be the end of the string, regardless of the extra space you allocated.

Next to that we should always make sure that the strings coming from the outside are valid UTF-8. So it should have been done here, but sadly wasn't.
```suggestion
	return StrMakeValid(name);
```